### DISCUSSION TITLE
Avalara plugin send the wrong pricing (net instead of gross) and then Avalara platform miscalculate taxes

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#9787] [General] [answered]
Hi,
I've noticed the tax calculation for  discounted and sales products is fixed now.

But there still a problem with Avalara because the plugin doesn't send the right price.
It looks like it sends the net price and not the gross then Avalara calculates taxes  in the wrong way.

An example:

**Saleor**
this product costs 90$.
The customer buy 2 items and then the total is 180$.
With taxes 193.95$.
<img width="871" alt="Screenshot 2022-05-18 at 09 02 39" src="https://user-images.githubusercontent.com/38940340/168978666-c0223558-8eee-4855-bdd9-6c86dce008ea.png">

**Avalara**
The plugin sends the net price 180$ instead of the gross price 193.95$
and then Avalara calculates the wrong amount of taxes (It subtract taxes from 180$)

<img width="1527" alt="Screenshot 2022-05-18 at 09 02 12" src="https://user-images.githubusercontent.com/38940340/168978680-c75ce47e-5898-4a84-b86a-d66f55beece1.png">

Let me know if is possible to have a fast fix (we can hardcode it)
or if we need to wait!

Really thanks in advance and kind regards,

Davide

**Comment by patrys** [comment]
Is your store configured to take gross or net prices? Can you provide step-by-step instructions to reproduce the issue?

**Comment by DidoMarchet** [comment]
Hi @patrys,

- we have configurate the plugin

- we remove all check from the tax configuration
<img width="519" alt="Screenshot 2022-05-18 at 14 39 13" src="https://user-images.githubusercontent.com/38940340/169040905-8bc0ed68-b553-4b20-9bbd-d32a4df6e7f2.png">

- we have selected the right tax group for the product type
<img width="914" alt="Screenshot 2022-05-18 at 14 41 52" src="https://user-images.githubusercontent.com/38940340/169041132-0aa2e49e-542c-49ce-baa4-f0d2c529872c.png">

- we charge the taxes to the product
<img width="436" alt="Screenshot 2022-05-18 at 14 43 02" src="https://user-images.githubusercontent.com/38940340/169041308-34dc22b7-dc33-4bdd-9e25-98646059998c.png">

- When an order comes in ste storefront taxes are correctly calculated

- When we check Avalara, taxes are calculated on the net of the product. Seems like the plugin send the net price instead of the gross price.

I hope it could be helpfull.

Thanks and regards,

Davide

**Comment by IKarbowiak** [comment, marked as answer]
Hi @DidoMarchet,
I reproduced the problem and I am already working on fixing it.

**Reply by IKarbowiak** [reply]
@DidoMarchet, the fix is here: https://github.com/saleor/saleor/pull/9820 and it's already merged to master. Thank you for reporting the issue.

**Reply by DidoMarchet** [reply]
> @DidoMarchet, the fix is here: https://github.com/saleor/saleor/pull/9820 and it's already merged to master. Thank you for reporting the issue.

Hi! Thanks for fixing and for the support! Now I'm traveling but my workmate will add the fix! Thanks and kind regards, Davide !

**Reply by Anubiella** [reply]
Hi! 
Yes we've applied the fix and now taxes on Avalara and Saleor match, so it seems ok, thansk a lot!!

Thanks and regards,
Lorenzo
